### ISSUE TITLE
Network errors when trying to use the format commands

### ISSUE THREAD

**Issue by MichaelPereira** [cogcmd/format#11]
Hi,

As I was trying to use the `mist:ec2-find` and running into the following error due to the size of the response:

``````
cog_1       | 2016-08-31T20:13:23.0332  (Carrier.Messaging.Connection:105) [warn] Message potentially too long (112010 bytes)
cog_1       | 2016-08-31T20:13:23.0777  (Carrier.Messaging.Connection:63) [info] Connection #PID<0.1695.0> connected to message bus
cog_1       | 2016-08-31T20:13:23.0808  () [error] GenServer :slack_api terminating
cog_1       | ** (Poison.SyntaxError) Unexpected token: <
cog_1       |     (poison) lib/poison/parser.ex:56: Poison.Parser.parse!/2
cog_1       |     (poison) lib/poison.ex:83: Poison.decode!/2
cog_1       |     (cog) lib/cog/adapters/slack/api.ex:323: Cog.Adapters.Slack.API.call_api!/3
cog_1       |     (cog) lib/cog/adapters/slack/api.ex:96: Cog.Adapters.Slack.API.handle_call/3
cog_1       |     (stdlib) gen_server.erl:629: :gen_server.try_handle_call/4
cog_1       |     (stdlib) gen_server.erl:661: :gen_server.handle_msg/5
cog_1       |     (stdlib) proc_lib.erl:240: :proc_lib.init_p_do_apply/3
cog_1       | Last message: {:send_message, "C1YQ7R1EC", "```\n{\n  \"tags\": {\n
<Truncated Content>
cog_1       | 2016-08-31T20:13:24.0505  (Cog.Adapters.Slack.RTMConnector:70) [info] Ready. Slack username: deploy-dm-bot, userid: U1YQM2C10.
cog_1       | 2016-08-31T20:13:24.0779  (Cog.Adapters.Slack.API:89) [info] Ready. Response cache TTL is 60 seconds.
``````

@imbriaco suggested on slack to format the result to avoid sending too much content to Slack.
The problem is that any format command I try results in the bot sending the following response:

```

deploy-dm-botBOT [4:36 PM]  
An error has occurred.

At `2016-08-31T20:36:41Z`, @michaelpereira initiated the following pipeline, assigned the unique ID `c97d74d1a6b84f49901d5182119b3ad9`:

   ` mist:ec2-find --region="us-east-1" | format:head 1`

The pipeline failed planning the invocation:

  `format:head 1`

The specific error was:

   Whoops! An error occurred. /usr/lib/ruby/2.2.0/net/http.rb:879:in `initialize': getaddrinfo: Try again (SocketError)
    from /usr/lib/ruby/2.2.0/net/http.rb:879:in `open'
    from /usr/lib/ruby/2.2.0/net/http.rb:879:in `block in connect'
    from /usr/lib/ruby/2.2.0/timeout.rb:73:in `timeout'
    from /usr/lib/ruby/2.2.0/net/http.rb:878:in `connect'
    from /usr/lib/ruby/2.2.0/net/http.rb:863:in `do_start'
    from /usr/lib/ruby/2.2.0/net/http.rb:852:in `start'
    from /usr/lib/ruby/2.2.0/net/http.rb:583:in `start'
    from /home/bundle/.bundle/ruby/2.2.0/gems/cog-rb-0.1.9/lib/cog/service.rb:65:in `put_or_post'
    from /home/bundle/.bundle/ruby/2.2.0/gems/cog-rb-0.1.9/lib/cog/service.rb:50:in `put'
    from /home/bundle/.bundle/ruby/2.2.0/gems/cog-rb-0.1.9/lib/cog/services/memory.rb:12:in `replace'
    from /home/bundle/lib/cog_cmd/format/head.rb:12:in `run_command'
    from /home/bundle/.bundle/ruby/2.2.0/gems/cog-rb-0.1.9/lib/cog/command.rb:39:in `execute'
    from /home/bundle/.bundle/ruby/2.2.0/gems/cog-rb-0.1.9/lib/cog/bundle.rb:37:in `run_command'
    from /home/bundle/.bundle/ruby/2.2.0/gems/cog-rb-0.1.9/lib/cog.rb:17:in `bundle'
    from /home/bundle/cog-command:10:in `<main>'
exit status 1
```

```

deploy-dm-botBOT [4:55 PM]  
An error has occurred.

At `2016-08-31T20:55:03Z`, @michaelpereira initiated the following pipeline, assigned the unique ID `b7130eb83bfd4794b15f2f05cfe4b3ad`:

   ` mist:ec2-find --region="us-east-1" | format:tail 1`

The pipeline failed planning the invocation:

  `format:tail 1`

The specific error was:

   Whoops! An error occurred. /usr/lib/ruby/2.2.0/net/http.rb:879:in `initialize': getaddrinfo: Try again (SocketError)
    from /usr/lib/ruby/2.2.0/net/http.rb:879:in `open'
    from /usr/lib/ruby/2.2.0/net/http.rb:879:in `block in connect'
    from /usr/lib/ruby/2.2.0/timeout.rb:73:in `timeout'
    from /usr/lib/ruby/2.2.0/net/http.rb:878:in `connect'
    from /usr/lib/ruby/2.2.0/net/http.rb:863:in `do_start'
    from /usr/lib/ruby/2.2.0/net/http.rb:852:in `start'
    from /usr/lib/ruby/2.2.0/net/http.rb:583:in `start'
    from /home/bundle/.bundle/ruby/2.2.0/gems/cog-rb-0.1.9/lib/cog/service.rb:65:in `put_or_post'
    from /home/bundle/.bundle/ruby/2.2.0/gems/cog-rb-0.1.9/lib/cog/service.rb:46:in `post'
    from /home/bundle/.bundle/ruby/2.2.0/gems/cog-rb-0.1.9/lib/cog/services/memory.rb:8:in `accum'
    from /home/bundle/.bundle/ruby/2.2.0/gems/cog-rb-0.1.9/lib/cog/command.rb:30:in `accumulate_input'
    from /home/bundle/.bundle/ruby/2.2.0/gems/cog-rb-0.1.9/lib/cog/command.rb:38:in `execute'
    from /home/bundle/.bundle/ruby/2.2.0/gems/cog-rb-0.1.9/lib/cog/bundle.rb:37:in `run_command'
    from /home/bundle/.bundle/ruby/2.2.0/gems/cog-rb-0.1.9/lib/cog.rb:17:in `bundle'
    from /home/bundle/cog-command:10:in `<main>'
exit status 1
```

How can I check to what machine is this request made?

I'm running the cog bot with the `docker-compose.yml` provided [in this issue](https://github.com/operable/cog/issues/905#issuecomment-243855441) to avoid operable/cog#905 maybe there is a network configuration that is not properly set?

**Comment by MichaelPereira** [comment]
When using the `docker-compose.yml` files from the master branch of operable/cog, the problem is slightly different:

```

deploy-dm-botBOT [5:40 PM]  
An error has occurred.

At `2016-08-31T21:40:12Z`, @michaelpereira initiated the following pipeline, assigned the unique ID `9be049c9691a45249a9215ab1c056ef7`:

   ` mist:ec2-find --region="us-east-1" | format:tail 1`

The pipeline failed planning the invocation:

  `format:tail 1`

The specific error was:

   Whoops! An error occurred. /usr/lib/ruby/2.2.0/net/http.rb:879:in `initialize': Connection refused - connect(2) for "localhost" port 4002 (Errno::ECONNREFUSED)
    from /usr/lib/ruby/2.2.0/net/http.rb:879:in `open'
    from /usr/lib/ruby/2.2.0/net/http.rb:879:in `block in connect'
    from /usr/lib/ruby/2.2.0/timeout.rb:73:in `timeout'
    from /usr/lib/ruby/2.2.0/net/http.rb:878:in `connect'
    from /usr/lib/ruby/2.2.0/net/http.rb:863:in `do_start'
    from /usr/lib/ruby/2.2.0/net/http.rb:852:in `start'
    from /usr/lib/ruby/2.2.0/net/http.rb:583:in `start'
    from /home/bundle/.bundle/ruby/2.2.0/gems/cog-rb-0.1.9/lib/cog/service.rb:65:in `put_or_post'
    from /home/bundle/.bundle/ruby/2.2.0/gems/cog-rb-0.1.9/lib/cog/service.rb:46:in `post'
    from /home/bundle/.bundle/ruby/2.2.0/gems/cog-rb-0.1.9/lib/cog/services/memory.rb:8:in `accum'
    from /home/bundle/.bundle/ruby/2.2.0/gems/cog-rb-0.1.9/lib/cog/command.rb:30:in `accumulate_input'
    from /home/bundle/.bundle/ruby/2.2.0/gems/cog-rb-0.1.9/lib/cog/command.rb:38:in `execute'
    from /home/bundle/.bundle/ruby/2.2.0/gems/cog-rb-0.1.9/lib/cog/bundle.rb:37:in `run_command'
    from /home/bundle/.bundle/ruby/2.2.0/gems/cog-rb-0.1.9/lib/cog.rb:17:in `bundle'
    from /home/bundle/cog-command:10:in `<main>'
exit status 1
```

It looks my value for `$COG_HOST` which is currently `localhost` as I am using Docker for Mac instead of docker-machines is wrong. I will retry using my machine ip value for `$COG_HOST`

**Comment by MichaelPereira** [comment]
Yes, this was the issue. Once I've set `$COG_HOST` to my local network ip, this problem disappeared.

Maybe there should be a check that `$COG_HOST` is set to a value that's accessible from the containers to avoid this kind of problems in the future?

**Comment by MichaelPereira** [comment]
Closed in favor of operable/cog#939
